core/src/test/scala/sttp/client/testing/HttpTest.scala

```diff
@@ -283,14 +285,20 @@ trait HttpTest[F[_]]
       val req = basicRequest.head(uri"$endpoint/compress")
       req.send().toFuture().map { resp => resp.code shouldBe StatusCode.Ok }
     }
+//TODO investigate as it breaks the rest of the tests
```

Comment:
Uncommenting this test causes tests to hang

Reply:
no idea :D which test hangs? maybe it's something in js/native?

Reply:
From the tests' output it seems like the one with big timeout

Reply:
it hangs jvm tests. I didn't run native and js locally.